inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```diff
@@ -817,14 +820,51 @@ else if (annotationDocumentId != -1
         return annotateableDocuments;
     }
 
+    private List<LeafReaderContext> sortLeaves(List<LeafReaderContext> aLeaves,
+            IndexSearcher aSearcher, MtasSpanQuery aQuery)
+        throws IOException
+    {
+        List<Pair<LeafReaderContext, Long>> mapToDocId = new ArrayList<>();
+
+        SpanWeight spanweight = aQuery.rewrite(aSearcher.getIndexReader()).createWeight(aSearcher,
+                false, 0);
+
+        for (LeafReaderContext leafReaderContext : aLeaves) {
+            Spans spans = spanweight.getSpans(leafReaderContext, SpanWeight.Postings.POSITIONS);
+            SegmentReader segmentReader = (SegmentReader) leafReaderContext.reader();
+            List<Long> allDocIds = new ArrayList<>();
+            if (spans != null) {
+                while (spans.nextDoc() != Spans.NO_MORE_DOCS) {
+                    if (segmentReader.numDocs() == segmentReader.maxDoc()
+                            || segmentReader.getLiveDocs().get(spans.docID())) {
+                        Document document = segmentReader.document(spans.docID());
+
+                        String rawSourceDocumentId = document.get(FIELD_SOURCE_DOCUMENT_ID);
+                        if (rawSourceDocumentId == null) {
+                            continue;
+                        }
+
+                        allDocIds.add(Long.valueOf(rawSourceDocumentId));
+                    }
+                }
+            }
+            if (allDocIds.size() == 1) {
```

Comment:
If you test for the size to be one here, I assume that the loop before must collect exactly one result. But if that is the case, why collect into a list and not into a simple variable - and why not stop the loop when the first result has been found. Could you explain/code-comment how the algorithm is intended to work?